packages/SwingSet/src/kernel/vatManager/vat-warehouse.js

```diff
@@ -0,0 +1,257 @@
+// @ts-check
+import { assert } from '@agoric/assert';
+import { makeVatTranslators } from '../vatTranslator';
+
+/**
+ * @param { ReturnType<typeof import('../state/kernelKeeper').default> } kernelKeeper
+ * @param { ReturnType<typeof import('../loadVat').makeVatLoader> } vatLoader
+ * @param {{ sizeHint?: number }=} policyOptions
+ *
+ * @typedef { ReturnType<typeof import('@agoric/swing-store-simple').initSwingStore> } SwingStore
+ * @typedef {(syscall: VatSyscallObject) => ['error', string] | ['ok', null] | ['ok', Capdata]} VatSyscallHandler
+ * @typedef {{ body: string, slots: unknown[] }} Capdata
+ * @typedef { [unknown, ...unknown[]] } Tagged
+ * @typedef { { moduleFormat: string }} Bundle
+ */
+export function makeVatWarehouse(kernelKeeper, vatLoader, policyOptions) {
+  const { sizeHint = 20 } = policyOptions || {};
```

Comment:
Let's raise this default to 30, I think we had 25 vats on the most recent testnet, and I don't want to thrash with just the core vats. Or maybe 50 to be safe.

In the longer run, we should plumb this through from the initial options, maybe `kernelOptions`, after some thought as to whether it's a chain-wide thing or is safe to set locally.